synapse/events/__init__.py

```diff
@@ -415,6 +429,36 @@ def auth_event_ids(self) -> Sequence[str]:
         """
         return [e for e, _ in self._dict["auth_events"]]
 
+    def relation(self) -> Optional[_EventRelation]:
+        """
+        Parse the event's relation information.
+
+        Returns:
+            The event relation information, if it is valid. None, otherwise.
+        """
+        relation = self.content.get("m.relates_to")
+        if not relation or not isinstance(relation, collections.abc.Mapping):
+            # No relation information.
+            return None
+
+        # Relations must have a type and parent event ID.
+        rel_type = relation.get("rel_type")
+        if not isinstance(rel_type, str):
+            return None
+
+        parent_id = relation.get("event_id")
+        if not isinstance(parent_id, str):
+            return None
+
+        # Annotations have a key field.
+        aggregation_key = None
+        if rel_type == RelationTypes.ANNOTATION:
+            aggregation_key = relation.get("key")
+            if not isinstance(aggregation_key, str):
+                aggregation_key = None
+
+        return _EventRelation(parent_id, rel_type, aggregation_key)
```

Comment:
So, I think I would prefer this to be a free standing function, as I would quite like to drive the `EventBase` to be purely a representation of an event (... to make it easier for us to replace this in Rust...).

I don't have super strong opinions on it though

Reply:
That's fine w/ me. Another option would be to put it on the `RelationsHandler` (so you would do `relations_handler.get_event_relation(event)` or something...)